### ISSUE TITLE
BUG in StableDiffusion.inpaint() creates blurred edges on output image

### ISSUE THREAD

**Issue by BFauber** [keras-team/keras-cv#1158]
I can run `StableDiffusion.inpaint()`, but I have noticed a blurring that is 8 pixels deep, and exists on the left side and top of the image matrix, regardless of the input image. 

Changing the location of the mask will impact the presence of the blurring, as the blurring will occur only where the mask is present, but never on the right or bottom of the image. I cannot create an instance where the blurring is on the right side nor bottom of the image/matrix.

I have not been able to link this blurring bug to any args/parameters of the model, but it looks to be something within the code, possibly the `tf.nn.max_pool2d()` padding step in `inpaint()`.

Do others see this same error in the output image?

https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/stable_diffusion/stable_diffusion.py#L333

**Comment by bobqywei** [comment]
I did some investigating into this issue. Seems like the problem is actually in the pre-trained image encoder being used:

<img width="487" alt="Screenshot 2022-12-20 at 12 24 43 PM" src="https://user-images.githubusercontent.com/26168216/208728347-02fcad30-51db-480a-8960-42f57a2eeba4.png">

In the snippet above, I passed the initial image through: `out = decoder(encoder(in))` and the resulting image is identical but with edge artifacts on the top and left. Does anyone else have more insight on this?
